packages/shared/src/components/crypto.ts

```diff
@@ -0,0 +1,29 @@
+const base64ToBuf = (b: string) =>
+  Uint8Array.from(atob(b), (c) => c.charCodeAt(0));
+
+export const decrypt = async (
+  input: string,
+  key: string,
+  algorithmName = 'AES-CBC',
+  algorithmLength = 256,
+  subtle = window.crypto.subtle,
+): Promise<string> => {
+  if (!subtle) {
+    return input;
+  }
+  const keyObject = await subtle.importKey(
+    'raw',
+    new TextEncoder().encode(key),
+    { name: algorithmName, length: algorithmLength },
+    true,
+    ['encrypt', 'decrypt'],
+  );
+  const [iv, cipherText] = input.split(':');
+  const plainTextBuffer = await subtle.decrypt(
+    { name: algorithmName, iv: base64ToBuf(iv) },
+    keyObject,
+    base64ToBuf(cipherText),
+  );
+
+  return new TextDecoder().decode(plainTextBuffer);
```

Comment:
I think this should be okay, if needed one idea is to potentially decrypt on the SSR side if needed.